<?xml version="1.0"?>
<launch>
  <!-- Interface -->
  <arg name="num_of_skeletons" default="12"/>
  <arg name="using_real_data" default="false"/>

	<!-- Making stream for debugging -->
	<group if="$(arg using_real_data)">
		<arg name="using_numbers" default="1 10"/>
		<param name="using_numbers" value="$(arg using_numbers)"/>
		<node name="skeleton_integrator" pkg="tms_ss_kinect_v2" args="$(arg using_numbers)" type="skeleton_integrator" output="screen"/>
	</group>
	<group unless="$(arg using_real_data)">
		<node name="test_skeleton_pub" pkg="tms_ss_kinect_v2" type="test_skeleton_pub.py"/>
	</group>

	<!-- For drawing joint points -->
	<node name="skeleton_viewer" pkg="tms_ss_kinect_v2" type="skeleton_viewer"/>

  <!-- For visualize skeleton in rViz -->
  <param name="skeleton_description1"  textfile="$(find skeleton_description)/urdf/skeleton/model_002.urdf"/>
  <param name="skeleton_description2"  textfile="$(find skeleton_description)/urdf/skeleton/model_002.urdf"/>
  <param name="skeleton_description3"  textfile="$(find skeleton_description)/urdf/skeleton/model_002.urdf"/>
  <param name="skeleton_description4"  textfile="$(find skeleton_description)/urdf/skeleton/model_002.urdf"/>
  <param name="skeleton_description5"  textfile="$(find skeleton_description)/urdf/skeleton/model_002.urdf"/>
  <param name="skeleton_description6"  textfile="$(find skeleton_description)/urdf/skeleton/model_002.urdf"/>
  <param name="skeleton_description7"  textfile="$(find skeleton_description)/urdf/skeleton/model_002.urdf"/>
  <param name="skeleton_description8"  textfile="$(find skeleton_description)/urdf/skeleton/model_002.urdf"/>
  <param name="skeleton_description9"  textfile="$(find skeleton_description)/urdf/skeleton/model_002.urdf"/>
  <param name="skeleton_description10" textfile="$(find skeleton_description)/urdf/skeleton/model_002.urdf"/>
  <param name="skeleton_description11" textfile="$(find skeleton_description)/urdf/skeleton/model_002.urdf"/>
  <param name="skeleton_description12" textfile="$(find skeleton_description)/urdf/skeleton/model_002.urdf"/>
  <param name="num_of_skeletons" value="$(arg num_of_skeletons)"/>
  <param name="using_db" value="false"/>
  <node name="skeleton_state_publisher" pkg="skeleton_description" type="skeleton_state_publisher" output="screen"/>
</launch>

